stdlib/gc.mli

```diff
@@ -458,15 +458,40 @@ module Memprof :
        type of records passed to the callback triggered by the
        sampling of an allocation. *)
 
+    type ('minor, 'major) tracker = {
+      alloc_minor: allocation -> 'minor option;
+      alloc_major: allocation -> 'major option;
+      promote: 'minor -> 'major option;
+      dealloc_minor: 'minor -> unit;
+      dealloc_major: 'major -> unit;
+    }
+    (**
+       A [('minor, 'major) tracker] describes how memprof should track
+       sampled blocks over their lifetime, keeping a user-defined piece
+       of metadata for each of them: ['minor] is the type of metadata
+       to keep for minor blocks, and ['major] the type of metadata
+       for major blocks.
+
+       If an allocation-tracking or promotion-tracking function returns [None],
+       memprof stops tracking the corresponding value.
+     *)
+
+    val null_tracker: ('minor, 'major) tracker
+    (** Default callbacks simply return [None] or [()] *)
+
+    val alloc_tracker: (allocation -> unit) -> ('a, 'a) tracker
+    (** Simple tracker that simply calls a function on each sampled
+        allocation. *)
+
+    val alloc_dealloc_tracker : (allocation -> 'a) -> ('a -> unit) -> ('a, 'a) tracker
+    (** Simple tracker that keeps some data of type ['a]
+        for each allocation, until the corresponding deallocation. *)
```

Comment:
Possibly worth putting these in a module of their own? e.g. `Gc.Memprof.Trackers` or something?

Reply:
(I ended up removing all of them except `null_tracker`.)